File: Facturacion/Clases/MySession.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

/// <summary>
/// Esta clase contiene el Objeto Session del usuario.
/// Definimos las propiedades que necesitamos para utilizarse tanto como en los forms .aspx u otras clases .vb.
/// </summary>
public class MySession
{
    //Properties
    public string Username { get; set; }
    public int? CodigoUser { get; set; }
    public int? CodigoRol { get; set; }
    public int? CodigoSesion { get; set; }
    public int? CodigoPais { get; set; }
    public int? CodigoEmpresa { get; set; }
    public int? CodigoPuesto { get; set; }

    private MySession()
    {
        Username = HttpContext.Current.Session["Username"] != null ? HttpContext.Current.Session["Username"].ToString() : null;
        CodigoUser = HttpContext.Current.Session["CodigoUser"] ? Convert.ToInt32(HttpContext.Current.Session["CodigoUser"]) : null;
        CodigoRol = HttpContext.Current.Session["CodigoRol"] != null ? Convert.ToInt32(HttpContext.Current.Session["CodigoRol"]) : null;
        CodigoSesion = HttpContext.Current.Session["CodigoSesion"] != null ? Convert.ToInt32(HttpContext.Current.Session["CodigoSesion"]) : null;
        CodigoPais = HttpContext.Current.Session["CodigoPais"] != null ? Convert.ToInt32(HttpContext.Current.Session["CodigoPais"]) : null;
        CodigoEmpresa = HttpContext.Current.Session["CodigoEmpresa"] != null ? Convert.ToInt32(HttpContext.Current.Session["CodigoEmpresa"]) : null;
        CodigoPuesto = HttpContext.Current.Session["CodigoPuesto"] != null ? Convert.ToInt32(HttpContext.Current.Session["CodigoPuesto"]) : null;

    }

    public static MySession Current
    {
        get
        {
            MySession session = (MySession)HttpContext.Current.Session["__MySession__"];
            if (session == null)
            {
                session = new MySession();
                HttpContext.Current.Session["__MySession__"] = session;
            }
            return session;
        }
    }

    public string GetWelcomeMessage()
    {
        if (!string.IsNullOrEmpty(this.Username))
            return "Bienvenido " + this.Username;
        else
            return "Bienvenido";

    }

}

File: Facturacion/App_Code/MySession.vb
Imports System.Web

Namespace HelperClasses
    Public Class MySession
        ' Properties
        Public Property Username As String
        Public Property CodigoUser As Integer
        Public Property CodigoRol As Integer
        Public Property CodigoSesion As Integer
        Public Property CodigoPais As Object
        Public Property CodigoEmpresa As Object
        Public Property CodigoPuesto As Object

        Private Sub New()
            Username = HttpContext.Current.Session("Username").ToString()
            CodigoUser = Convert.ToInt32(HttpContext.Current.Session("CodigoUser"))
            CodigoRol = Convert.ToInt32(HttpContext.Current.Session("CodigoRol"))
            CodigoSesion = Convert.ToInt32(HttpContext.Current.Session("CodigoSesion"))
            CodigoPais = Convert.ToInt32(HttpContext.Current.Session("CodigoPais"))
            CodigoEmpresa = Convert.ToInt32(HttpContext.Current.Session("CodigoEmpresa"))
            CodigoPuesto = Convert.ToInt32(HttpContext.Current.Session("CodigoPuesto"))
        End Sub

        ' Get Current Session
        Public Shared ReadOnly Property Current As MySession
            Get
                Dim session As MySession = DirectCast(HttpContext.Current.Session("__MySession__"), MySession)

                If session Is Nothing Then

                    session = New MySession()
                    HttpContext.Current.Session("__MySession__") = session

                End If

                Return session
            End Get
        End Property
    End Class
End Namespace
